### ISSUE TITLE
Source for the Python parser

### ISSUE THREAD

**Issue by MatthieuDartiailh** [we-like-parsers/pegen#9]
pegen/data contains a Python parser supposedly generated from data/python.gram but this file only contains C like rules ? Would it be possible to get the file containing the python rules ?

**Comment by pablogsal** [comment]
Unfortunately, there isn't a version of that grammar with Python rules, this needs to be written and the helpers and other stuff needs to be ported from the C files.

**Comment by MatthieuDartiailh** [comment]
I would be interested in any pointer you may have as the best way to get started then. I would like to port the parser of enaml which is based on ply (https://github.com/nucleic/enaml/tree/main/enaml/core/parser). Enaml defines a superset of Python so I need a full Python parser and I would be happy to contribute it to pegen.

**Comment by pablogsal** [comment]
> Enaml defines a superset of Python so I need a full Python parser and I would be happy to contribute it to pegen.

That would be great. My advice would be:

* Start with the file here: https://github.com/python/cpython/blob/master/Grammar/python.gram
* Remove all the C types (but not the actions for now).
* Identify the actions that create ast nodes and substitute them for Python nodes in the ast module. For instance:

```
while_stmt[stmt_ty]:
    | 'while' a=named_expression &&':' b=block c=[else_block] { _Py_While(a, b, c, EXTRA) }
```

will become (not real final code):

```
while_stmt:
    | 'while' a=named_expression &&':' b=block c=[else_block] { ast.While(a, b, c) }
```
* Port the rest of the actions to propagate accordingly the values. This is much easier than in C because you can use python code and types in the actions. For example `_PyPegen_singleton_seq(p, c)` could be simply `[c]` in Python code.

* Port all the error-checking helpers to Python (these are the `CHECK` macros and the `raise...` functions.

* Once this is done, there is still some work to make sure line numbers and column numbers are correct, but we can discuss how to fix those later.

**Comment by MatthieuDartiailh** [comment]
Sounds good. I was a bit worried about the line numbers but if we can start without I will go with that.

**Comment by MatthieuDartiailh** [comment]
Closed by #11
